app/assets/javascripts/lookup_keys.js

```diff
@@ -217,3 +226,61 @@ function validatorTypeSelected(item){
   var validator_rule_field = $(item).closest('.fields').find("[id$='_validator_rule']");
   validator_rule_field.attr('disabled', validatorType == "" ? 'disabled' : null);
 }
+
+const KEY_DELM = ",";
+const EQ_DELM  = "=";
+
+function match_to_key_value(match){
+  var regex = new RegExp('[' + KEY_DELM + EQ_DELM + ']');
+
+  var keys = [], values = [],
+  split_matcher = match.replace(/(\s+,) | (,\s+)/g, '').split(regex);
+
+  $.each(split_matcher, function (index, value) {
+    if (index % 2 === 0) {
+      keys.push(value);
+    } else {
+      values.push(value);
+    }
+  });
+
+  return [keys.join(KEY_DELM), values.join(KEY_DELM)];
+}
+
+function key_value_to_match(keys, values){
+  var match = "";
+  keys.split(KEY_DELM).forEach(function (el, index) {
+   match += el + EQ_DELM + values.split(KEY_DELM)[index] + KEY_DELM;
+  });
+
+  return match.slice(0, -1);
+}
+
+function fill_in_matchers(){
+  $('.matchers').each(function () {
+    var matcher = $(this);
+    var match = matcher.find('.match').val();
+    var matcher_key = matcher.find('.matcher_key');
+    var matcher_value = matcher.find('.matcher_value');
+    var order = matcher.closest('.matcher-parent').find('#order').val().split('\n');
+    matcher_key.empty();
+    $.each(order, function (index, value) {
+      matcher_key.append($("<option>", {value: encodeURI(value), html: encodeURI(value)}));
```

Comment:
Yeah, I flagged this earlier on, I think @orrabin was struggling to find a way to escape the HTML (as if you don't it allows script injection) and URI encoding was the closest?